addon/core/consumer.js

```diff
@@ -8,12 +9,17 @@ export default EmberObject.extend({
 
   init() {
     this._super(...arguments);
-    set(this,'subscriptions', Subscriptions.create(getOwner(this).ownerInjection(), { consumer: this }));
-    set(this,'connection', Connection.create(getOwner(this).ownerInjection(), { consumer: this }));
+    this.subscriptions = Subscriptions.create(getOwner(this).ownerInjection(), { consumer: this });
+    this.connection = Connection.create(getOwner(this).ownerInjection(), { consumer: this });
   },
 
   send(data) {
-    get(this,'connection').send(data);
-  }
+    this.connection.send(data);
+  },
 
+  willDestroy() {
+    this._super();
+    run(this.connection, 'destroy');
```

Comment:
We need to do our own cleaning up since these are created outside of the container.  When the instance of the service is destroyed we need to signal to all these object we created to also destroy.  This way their internal timers also are stopped.